src/robot/client.ts

```diff
@@ -336,6 +341,36 @@ export class RobotClient implements Robot {
          * read and then write to 'peerConn', even after we have awaited/paused.
          */
         this.peerConn = webRTCConn.peerConnection; // eslint-disable-line require-atomic-updates
+        this.peerConn.addEventListener('iceconnectionstatechange', () => {
+          /*
+           * TODO: are there any disconnection scenarios where we can reuse the
+           * same connection and restart ice?
+           *
+           * All connection loss scenarios I tested seem to result in the peer
+           * connection getting closed, so restarting ice is not a valid way to
+           * recover.
+           */
+          if (this.peerConn?.iceConnectionState === 'closed') {
```

Comment:
you may want to only do this if we knew we 'connected' before

Reply:
unfortunately we can't get the previous state from the event. we could add a variable to track this, but maybe it's enough to rely on the fact that we set this listener right after we get the peer connection. we only reconnect on `close` so we know that peer connection will not accepts more requests after that.